### PR TITLE
feat: New Lettermark colors

### LINKED ISSUE #9867: Adding colour pairs for lettermarks on the app
## Is your feature request related to a problem?

Currently we only have one letter mark design on our app for both organisations and profiles. While this is fine it feels a little dull and lacks colour. 

Frame 9262

## Describe the solution you'd like

We propose adding in 8 new designs to the product to brighten up the user experience. All the colours pass the Contrast Checker for text for : 
- AA Large
- AAA Large
- AA Normal

These colours are not used in the app anymore since Chris updated the new data palette but they will still seem familiar to the user. This is also an advantage as the letter marks will have their own unique colour scheme and not confuse the user between graph colours or primary action colours. 

Frame 9624

Frame 9271

## Describe alternatives you've considered

I did experiment around with using the new data colours but firstly realised that this may clash with our branding that we have already - too bold and punchy, and secondly that 80% of the colour combinations that worked did not pass the Contrast Checker. 

## Additional context

[Contrast Checker ](https://colourcontrast.cc/ff906e/2a3d65)

[Why Colour Contrast Matters](https://accessibility.blog.gov.uk/2016/06/17/colour-contrast-why-does-it-matter/)

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

There are some lovely, accessible colors for Lettermarks and Profile Avatars so we should use them!

Closes https://github.com/PostHog/posthog/issues/9867

## Changes

* ProfileBubbles
Screenshot 2022-08-05 at 16 33 29
* Lettermarks
Screenshot 2022-08-05 at 16 33 37

👉 *Stay up-po-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by lottiecoxon — APPROVED** [review]
Looks great Ben!

File: frontend/src/lib/components/Lettermark/Lettermark.tsx [hunk 1]
```diff
@@ -3,30 +3,41 @@ import React from 'react'
...
-    Default = 'default',
-    Gray = 'gray',
+    Default = 1,
+    Gray = 8,
 }
 
 export interface LettermarkProps {
     /** Name or value the lettermark should represent. */
     name?: string | number | null
+    // If given, will choose a color based on the index
```
Comment (Twixes) [inline frontend/src/lib/components/Lettermark/Lettermark.tsx:13]: Tip: Use JSDoc syntax `/** Lorem ipsum */` for these comments, that way they get helpfully shown in tooltips by IDEs

File: frontend/src/lib/components/Lettermark/Lettermark.tsx [hunk 2]
```diff
@@ -3,30 +3,41 @@ import React from 'react'
...
  */
-export function Lettermark({ name, color = LettermarkColor.Default }: LettermarkProps): JSX.Element {
+export function Lettermark({ name, index, color, rounded = false }: LettermarkProps): JSX.Element {
     const representation = name
         ? typeof name === 'number'
             ? String(Math.floor(name))
             : name.toLocaleUpperCase().charAt(0)
         : '?'
 
+    const colorIndex = color ? color : typeof index === 'number' ? (index % 8) + 1 : undefined
```
Comment (Twixes) [inline frontend/src/lib/components/Lettermark/Lettermark.tsx:32]: 8 looka like kind of a magic number, can we make it explicitly a `CONSTANT`?

File: frontend/src/lib/components/Lettermark/Lettermark.tsx [hunk 3]
```diff
@@ -3,30 +3,41 @@ import React from 'react'
 import './Lettermark.scss'
 
 export enum LettermarkColor {
-    Default = 'default',
-    Gray = 'gray',
+    Default = 1,
+    Gray = 8,
```
Comment (Twixes) [inline frontend/src/lib/components/Lettermark/Lettermark.tsx:7]: This isn't really gray, just one of the colored variants. This change doesn't look great in funnels for instance:

Zrzut ekranu 2022-08-8 o 13 12 49

Can we keep the existing gray as a variant?

**Review by Twixes — CHANGES_REQUESTED** [review]
A few notes, but overall this is really nice!

File: frontend/src/lib/components/Lettermark/Lettermark.tsx (hunk 3 above)
Comment (benjackwhite) [reply frontend/src/lib/components/Lettermark/Lettermark.tsx:7]: ah good catch. Sure I'll add it in as a unique variant perhaps...

**Review by Twixes — APPROVED** [review]
𝓕𝓪𝓷𝓬𝔂

**Comment by clarkus** [comment]
I was in the middle of reviewing this when it was merged. Does this change affect the use of gravatar avatars for users who have that set up?

**Comment by Twixes** [comment]
I don't believe it does at all @clarkus

**Pull request merged**